plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergModule.java

```diff
@@ -130,10 +130,18 @@ public void configure(Binder binder)
         newOptionalBinder(binder, IcebergFileSystemFactory.class).setDefault().to(DefaultIcebergFileSystemFactory.class).in(Scopes.SINGLETON);
         newOptionalBinder(binder, CacheKeyProvider.class).setBinding().to(IcebergCacheKeyProvider.class).in(Scopes.SINGLETON);
 
+        closingBinder(binder).registerExecutor(ExecutorService.class);
```

Comment:
does this not require a binding annotation? What is the behaviour when mixing both annotated and unannotated bindings for same type? ie. can an unannotated injection be satisfied using an annotated binding. If yes then that's not desirable.

Reply:
Based on [Guice Mental Model](https://github.com/google/guice/wiki/MentalModel#guice-keys), my understanding is that `Key.of(ExecutorService.class)` and `Key.of(ExecutorService.class, SomeAnnotation.class)` are not equal. So, an annotated binding cannot be injected into an unannotated parameter, and vice versa.

Reply:
Yes, seems true. I tried it out with a test. It does seem "out of place" though. No change requested.